pkg/provisioner/ironic/ironic.go

```diff
@@ -1645,27 +1645,26 @@ func (p *ironicProvisioner) Deprovision(force bool) (result provisioner.Result,
 		)
 
 	case nodes.CleanFail:
```

Comment:
We need to treat this like the Error state and do:

```
if !force {
    return operationFailed(fmt.Sprintf("Cleaning failed: %s", ironicNode.LastError))
}
```

(Speaking of which, it looks like l. 1638 is the last remaining place where we're not using the semantic `operation*()` functions for return values.)